nifi-nar-bundles/nifi-aws-bundle/nifi-aws-processors/src/main/java/org/apache/nifi/processors/aws/s3/ListS3.java

```diff
@@ -267,26 +267,28 @@ public void onTrigger(final ProcessContext context, final ProcessSession session
             commit(context, session, listCount);
             listCount = 0;
         } while (bucketLister.isTruncated());
-        currentTimestamp = maxTimestamp;
+
+        if (maxTimestamp > currentTimestamp) {
+            currentTimestamp = maxTimestamp;
+        }
```

Comment:
`maxTimestamp` should always be greater than `currentTimestamp`, but this adds a sanity check.